features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ctxspend/CTXSpendViewModel.kt

```
@@ -272,4 +272,14 @@ class CTXSpendViewModel @Inject constructor(
     fun logEvent(eventName: String) {
         analytics.logEvent(eventName, mapOf())
     }
+
+    suspend fun needToCheckRefreshToken() = repository.needToCheckRefreshToken()
+
+    suspend fun checkToken(): Boolean {
+        return try {
+            repository.refreshToken()
+        } catch (ex: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)